mainProject/script.js

```diff
@@ -0,0 +1,459 @@
+      publication.author === undefined ||
+      publication.text === undefined ||
+      publication.createdAt === undefined ||
+      publication.id === "" ||
+      publication.author === "" ||
+      publication.text === "" ||
+      publication.createdAt === "" ||
+      publication.text.length > 280
+    ) {
+      return false;
+    } else {
+      return true;
+    }
+  }
+  get id() {
+    return this._id;
+  }
+
+  get author() {
+    return this._author;
+  }
+
+  get text() {
+    return this._text;
+  }
+
+  get createdAt() {
+    return this._createdAt;
+  }
+
+  set text(text) {
+    this._text = text;
+  }
+}
+
+class Comment extends Publication {
+  constructor(id, text, author) {
+    super(id, text, author);
+  }
+}
+
+class Tweet extends Publication {
+  constructor(id, text, author) {
+    super(id, text, author);
+    this._comments = [];
+  }
+
+  static validate(tweet) {
+    return super.validate(tweet) && Boolean(tweet.comments);
+  }
+
+  get comments() {
+    return this._comments;
+  }
+}
+
+class TweetsCollection {
+  constructor(arrayOfTweets) {
+    this._collectionOfTweets = [];
+    for (let tweet of arrayOfTweets) {
+      if (Tweet.validate(tweet)) {
+        this._collectionOfTweets.push(tweet);
+      }
+    }
+    this._user = undefined;
+  }
+
+  get user() {
+    return this._user;
+  }
+
+  set changeUser(user) {
+    this._user = user;
+  }
+
+  addAll(arrayOfTweets) {
+    let nonValidateTweets = [];
+    for (let tweet of arrayOfTweets) {
+      if (Tweet.validate(tweet)) {
+        this._collectionOfTweets.push(tweet);
+      } else {
+        nonValidateTweets.push(tweet);
+      }
+    }
+    return nonValidateTweets;
+  }
+
+  getPage(skip = 0, top = 10, filterConfig = {}) {
+    function sortFunction(a, b) {
+      let dateA = new Date(a.createdAt);
+      let dateB = new Date(b.createdAt);
+      return dateA > dateB ? 1 : -1;
+    }
+    this._collectionOfTweets.sort(sortFunction);
+    let arrayFilter = [];
+    let end = skip + top;
+
+    if (
+      filterConfig.author === undefined &&
+      filterConfig.dateFrom === undefined &&
+      filterConfig.dateTo === undefined &&
+      filterConfig.hashtags === undefined &&
+      filterConfig.text === undefined
+    ) {
+      if (skip + top > this._collectionOfTweets.length) {
+        end = this._collectionOfTweets.length;
+      }
+      return (arrayFilter = this._collectionOfTweets.slice(skip, end));
+    }
+
+    for (let i = 0, j = 0; i < this._collectionOfTweets.length; i++) {
+      if (filterConfig.author !== undefined) {
+        if (this._collectionOfTweets[i].author !== filterConfig.author) {
+          continue;
+        }
+      }
+      if (filterConfig.dateFrom !== undefined) {
+        if (this._collectionOfTweets[i].createdAt <= filterConfig.dateFrom) {
+          continue;
+        }
+      }
+      if (filterConfig.dateTo !== undefined) {
+        if (this._collectionOfTweets[i].createdAt >= filterConfig.dateTo) {
```

Comment:
!filterConfig.dateTo && this._collectionOfTweets[i].createdAt >= filterConfig.dateTo